src/Logging/Logging.Batching/BatchingLogServiceProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeMonkeys.Logging.Batching
+{
+    public abstract class BatchingLogServiceProvider<TOptions> : LogServiceProvider<TOptions>
+        where TOptions : BatchingLogOptions
+    {
+        private BlockingCollection<LogMessage> _queue;
+        private CancellationTokenSource _cts;
+
+        private TimeSpan _flushPeriod;
+        private readonly int? _queueCapacity;
+        private int? _batchCapacity;
+
+        protected BatchingLogServiceProvider(TOptions options) 
+            : base(options)
+        {
+            OnOptionsHasChanged(options);
+
+            _queueCapacity = options.QueueCapacity;
+
+            Run();
+        }
+
+        public override void ProcessMessage(LogMessage message)
+        {
+            if (_queue.IsAddingCompleted)
+                return;
+
+            try
+            {
+                _queue.Add(message, _cts.Token);
+            }
+            catch { }
+        }
+
+        protected override void OnOptionsHasChanged(TOptions options)
+        {
+            _flushPeriod = options.FlushPeriod;
+            _batchCapacity = options.BatchCapacity;            
+
+            var previousIsEnabled = IsEnabled;
```

Comment:
`wasEnabled` ? :D

Reply:
I used IsEnabled because it is the name of the property